common/src/main/java/io/homeassistant/companion/android/common/notifications/NotificationFunctions.kt

```diff
@@ -274,3 +290,101 @@ fun handleText(
         builder.setStyle(NotificationCompat.BigTextStyle().bigText(text))
     }
 }
+
+fun createAction(
+    context: Context,
+    notificationAction: NotificationAction,
+    builder: NotificationCompat.Builder,
+    pendingIntent: PendingIntent,
+    type: String
+) {
+    val remoteInput: RemoteInput = RemoteInput.Builder(NotificationData.KEY_TEXT_REPLY).run {
+        setLabel(context.getString(R.string.action_reply))
+        build()
+    }
+    val action = NotificationCompat.Action.Builder(
+        when (type) {
+            "action" -> R.drawable.ic_stat_ic_notification
+            "uri" -> R.drawable.ic_globe
+            else -> R.drawable.ic_baseline_reply_24
+        },
+        notificationAction.title,
+        pendingIntent
+    )
+    if (type == "reply")
+        action.addRemoteInput(remoteInput)
+    builder.addAction(
+        action.build()
+    )
+        .build()
+}
+
+fun createPendingIntent(
+    context: Context,
+    requestCode: Int,
+    intent: Intent,
+    type: String

```

Comment:
As this is only used as a flag to check if it is a reply or not a boolean would be better. If you have future plans use an enum or constants :)